ext/datasource/plugin/flow_rule.go

```diff
@@ -0,0 +1,47 @@
+package plugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/alibaba/sentinel-golang/core/flow"
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/pkg/errors"
+)
+
+// FlowRulesJsonConverter provide JSON  as the default serialization for flow.FlowRule
+func FlowRulesJsonConverter(src []byte) (interface{}, error) {
+	if len(src) == 0 {
+		return nil, nil
+	}
+	ret := make([]flow.FlowRule, 0)
+	err := json.Unmarshal(src, &ret)
+	if err != nil {
+		return nil, errors.Errorf("Fail to unmarshal source:%+v to []flow.FlowRule, err:%+v", src, err)
+	}
+	return ret, nil
+}
+
+// FlowRulesUpdater load the newest []flow.FlowRule to downstream flow component.
+func FlowRulesUpdater(data interface{}) error {
+	rules := make([]*flow.FlowRule, 0)
+	if data == nil {
+		rules = nil
+	} else {
+		val, ok := data.([]flow.FlowRule)
+		if !ok {
+			return errors.New(fmt.Sprintf("Fail to type assert data to []flow.FlowRule, in fact, data: %+v", data))
+		}
+		for _, v := range val {
+			rules = append(rules, &v)
+		}
+	}
+	succ, err := flow.LoadRules(rules)
+	if succ {
```

Comment:
The `bool` returned by `LoadRules()` does not indicate error, so maybe we could just return `err` here?

Reply:
make sense